include/deal.II/dofs/dof_accessor.templates.h

```diff
@@ -3123,6 +3123,242 @@ namespace internal
 
 
 
+      template <int dim, int spacedim, bool level_dof_access>
+      static bool
+      p_refine_flag_set(const DoFCellAccessor<DoFHandler<dim, spacedim>,
+                                              level_dof_access> &accessor)
+      {
+        (void)accessor;
+        // ::DoFHandler does not support p-refinement
+        Assert(
+          false,
+          (typename std::decay<decltype(accessor)>::type::ExcInvalidObject()));
+        return false;
+      }
+
+
+
+      template <int dim, int spacedim, bool level_dof_access>
+      static bool
+      p_refine_flag_set(
+        const DoFCellAccessor<dealii::hp::DoFHandler<dim, spacedim>,
+                              level_dof_access> &accessor)
+      {
+        Assert(
+          accessor.dof_handler != nullptr,
+          (typename std::decay<decltype(accessor)>::type::ExcInvalidObject()));
+        Assert(static_cast<unsigned int>(accessor.level()) <
+                 accessor.dof_handler->levels.size(),
+               ExcMessage("DoFHandler not initialized"));
+
+        const auto &p_refine_flags =
+          accessor.dof_handler->levels[accessor.level()]->p_refine_flags;
+        Assert(static_cast<unsigned int>(accessor.present_index) <
+                 p_refine_flags.size(),
+               ExcIndexRange(accessor.present_index, 0, p_refine_flags.size()));
+
+        return p_refine_flags[accessor.present_index];
+      }
+
+
+
+      template <int dim, int spacedim, bool level_dof_access>
+      static void
+      set_p_refine_flag(const DoFCellAccessor<DoFHandler<dim, spacedim>,
+                                              level_dof_access> &accessor)
+      {
+        (void)accessor;
+        // ::DoFHandler does not support p-refinement
+        Assert(
+          false,
+          (typename std::decay<decltype(accessor)>::type::ExcInvalidObject()));
+      }
+
+
+
+      template <int dim, int spacedim, bool level_dof_access>
+      static void
+      set_p_refine_flag(
+        const DoFCellAccessor<dealii::hp::DoFHandler<dim, spacedim>,
+                              level_dof_access> &accessor)
+      {
+        Assert(
+          accessor.dof_handler != nullptr,
+          (typename std::decay<decltype(accessor)>::type::ExcInvalidObject()));
+        Assert(static_cast<unsigned int>(accessor.level()) <
+                 accessor.dof_handler->levels.size(),
+               ExcMessage("DoFHandler not initialized"));
+
+        auto &p_refine_flags =
+          accessor.dof_handler->levels[accessor.level()]->p_refine_flags;
+        Assert(static_cast<unsigned int>(accessor.present_index) <
+                 p_refine_flags.size(),
+               ExcIndexRange(accessor.present_index, 0, p_refine_flags.size()));
+
+        p_refine_flags[accessor.present_index] = true;
```

Comment:
what happens if the coarsen flag is currently set? we should throw an exception in that case.

same for when you want to set the coarsen flag and the refine flag is already set.

Reply:
These functions only depict implementation. We check for flags that have been already set in the `DoFCellAccessor` member functions (below).

Reply:
Yes, I think I found that later as well. This is ok then.